packages/options/options.pony

```diff
@@ -101,6 +101,14 @@ class Options is Iterator[(ParsedOption | ParseError | None)]
       _arguments.push(arg.clone())
     end
 
+  new from_array(args: Array[String val] box, fatal: Bool = true) =>
```

Comment:
Hmmm I see that it is storing args as String ref already. That seems like a very odd choice. I'm not sure what the purpose of making them `ref` rather than `val` would be.